### ISSUE TITLE
[2.5.0] catch is now called catch2

### ISSUE THREAD

**Issue by tuxmaster5000** [pybind/pybind11#2168]
To use the system catch lib for testing, a little patch is needed.
```
diff -Nuar pybind11-2.5.0.org/tools/FindCatch.cmake pybind11-2.5.0/tools/FindCatch.cmake
--- pybind11-2.5.0.org/tools/FindCatch.cmake	2020-04-16 07:33:50.871360486 +0200
+++ pybind11-2.5.0/tools/FindCatch.cmake	2020-04-16 08:14:30.524482052 +0200
@@ -38,7 +38,7 @@
 endfunction()
 
 # Look for catch locally
-find_path(CATCH_INCLUDE_DIR NAMES catch.hpp PATH_SUFFIXES catch)
+find_path(CATCH_INCLUDE_DIR NAMES catch.hpp PATH_SUFFIXES catch2)
 if(CATCH_INCLUDE_DIR)
   _get_catch_version()
 endif()
```

**Comment by bstaletic** [comment]
Would you mind making a pull reequest?

**Comment by YannickJadoul** [comment]
Fixed in #2299. Thanks again, @tuxmaster5000!
